src/sage/matroids/basis_exchange_matroid.pyx

```diff
@@ -1957,8 +1956,8 @@ cdef class BasisExchangeMatroid(Matroid):
             else:
                 k = min(self.full_rank() - 1, 2)
                 fie, f_vec = self._flat_element_inv(k)
-                self._weak_invariant_var = hash(tuple([tuple([(f, len(fie[f])) for f in sorted(fie)]), f_vec]))
-                self._weak_partition_var = SetSystem(self._E, [fie[f] for f in sorted(fie)])
+                self._weak_invariant_var = hash(tuple([tuple([(f, len(fie[f])) for f in sorted(fie, key=str)]), f_vec]))
+                self._weak_partition_var = SetSystem(self._E, [fie[f] for f in sorted(fie, key=str)])
```

Comment:
Why not let the sorting use the natural way? This makes `10` appear before `2`. If you can a case where this fails, a doctest should be added. In that case, I think a `try`-`except` block falling back to `str` sorting is better (assuming that uniquely determines things).

Reply:
Oops, I didn't think of that. I want to allow the groundset to have elements of mixed type (integers and strings). I have fixed it now.